src/charm.py

```diff
@@ -174,14 +174,47 @@ def _update_workload_container_config(self, event):
         try:
             upgrade_required = self.migration_is_required(schema_container, dsn)
         except Exception as e:
-            LOGGER.error("Failed to determe if schema upgrade required.")
+            LOGGER.error(f"Failed to determe if schema upgrade required: {e}")
             self.unit.status = WaitingStatus("Unable to determine database readiness")
-            LOGGER.error(e)
             event.defer()
             return
+
         if upgrade_required:
             self.schema_upgrade(schema_container, dsn)
 
+        return
+
+    def get_env_vars(self) -> dict:
```

Comment:
Where did these changes come from?

Reply:
that was me trying to clean things up. Pylint fails because the function is too long and complex, has too much return statements, so I started moving parts of it into different functions